### PR TITLE
User auth

### PR DESCRIPTION
**Pull request by sparksofl**
Implemented basic user auth using devise gem.

### CODE REVIEW

**Comment by brytannia** [comment]
Please, in future, use more self-explanatory messages for commits.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
-  # Prevent CSRF attacks by raising an exception.
-  # For APIs, you may want to use :null_session instead.
+  before_action :configure_permitted_parameters, if: :devise_controller?
+  
   protect_from_forgery with: :exception
+
+  protected
+
+  def configure_permitted_parameters
+  	devise_parameter_sanitizer.for(:sign_up) << :first_name
```
Comment (brytannia) [inline app/controllers/application_controller.rb]: this method doesn't look dry. Please, consider this article about ruby array to find out how this method can be rewritten http://docs.ruby-lang.org/en/2.0.0/Array.html#method-i-3C-3C

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class UsersController < ApplicationController
```
Comment (brytannia) [inline app/controllers/users_controller.rb:1]: according to ruby style guide, better to change tabs with 2 whitespaces

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
```
Comment (brytannia) [inline app/models/user.rb]: please, delete this comment. it's redundant

File: app/views/devise/confirmations/new.html.slim [hunk 4]
```diff
@@ -0,0 +1,11 @@
+h2
```
Comment (brytannia) [inline app/views/devise/confirmations/new.html.slim]: please, align this and others view files

Comment (brytannia) [inline app/views/devise/confirmations/new.html.slim]: if you don't need some custom devise view, you can delete it from folder /views/device... and device will choose standard one.

**Comment by brytannia** [comment]
@sparksofl I've reviewed PR. Ask me questions if you have some or just add required changes. :octocat:

**Comment by sparksofl** [comment]
@brytannia I've added changes. Would you check?

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,5 +1,12 @@
...
-  # Prevent CSRF attacks by raising an exception.
-  # For APIs, you may want to use :null_session instead.
+  before_action :configure_permitted_parameters, if: :devise_controller?
+  
   protect_from_forgery with: :exception
+
+  protected
+
+  def configure_permitted_parameters
+  	temp = [:first_name, :last_name, :email, :password, :password_confirmation]
```
Comment (brytannia) [inline app/controllers/application_controller.rb]: nice solution! however I would recommend to rewrite this as
[:first_name, :last_name, :email, :password, :password_confirmation].each do |field|
    devise_parameter_sanitizer.for(:sign_up) << field
end

How do you think? Which syntax do you like best?

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (brytannia) [inline app/controllers/users_controller.rb:1]: did you removed tabs from this file?

File: config/secrets.yml.example [hunk 6]
```diff
@@ -1,22 +0,0 @@
-# Be sure to restart your server when you modify this file.
```
Comment (brytannia) [inline config/secrets.yml.example:1]: why did you delete this file from repo?

File: config/routes.rb [hunk 7]
```diff
@@ -1,4 +1,8 @@
 Rails.application.routes.draw do
+  devise_for :users
+
+  resources :users
+  root :to => 'users#index'
```
Comment (brytannia) [inline config/routes.rb]: please, use new hash style
 to:  'users#index'

File: config/database.yml.example [hunk 8]
```diff
@@ -1,85 +0,0 @@
-# PostgreSQL. Versions 8.2 and up are supported.
```
Comment (brytannia) [inline config/database.yml.example:1]: please, return db example file :)

File: app/views/layouts/application.html.slim [hunk 9]
```diff
@@ -0,0 +1,19 @@
+doctype html
```
Comment (brytannia) [inline app/views/layouts/application.html.slim:1]: tabs are used only in js files. In .rb, .erb, .slim .haml whitespaces are used.

**Comment by brytannia** [comment]
@sparksofl I've added some other comments about new changes and old ones I missed first time. Please, check them out.

File: app/controllers/application_controller.rb (hunk 5 above)
Comment (sparksofl) [inline app/controllers/application_controller.rb]: According to http://stackoverflow.com/questions/2122380/using-do-block-vs-brackets
because there is a small operation such as pushing field to an array, I prefer brackets. But I'm not sure about it.

Comment (brytannia) [inline app/controllers/application_controller.rb]: yes, in general that's right. however devise_parameter_sanitizer.for(:sign_up) << p so it goes well in block and as bonus you don't have to create additional temporal variable. Anyway, choose what you want. It's all yours :)

**Comment by brytannia** [comment]
@sparksofl I've cleaned up schema file and view application_layout. Now everything looks good.
Congratulations for your first accepted PR! :tada:
